--- mutt/init.c	2009-12-14 19:24:59.000000000 +0100
+++ mutt/init.c.new	2016-10-19 16:00:26.065999981 +0200
@@ -2928,9 +2928,8 @@ void mutt_init (int skip_sys_rc, LIST *c
 #define DOMAIN buffer
   if (!p && getdnsdomainname (buffer, sizeof (buffer)) == -1)
     Fqdn = safe_strdup ("@");
-  else
 #endif /* DOMAIN */
-    if (*DOMAIN != '@')
+  if (strlen(DOMAIN) > 0 && *DOMAIN != '@')
   {
     Fqdn = safe_malloc (mutt_strlen (DOMAIN) + mutt_strlen (Hostname) + 2);
     sprintf (Fqdn, "%s.%s", NONULL(Hostname), DOMAIN);	/* __SPRINTF_CHECKED__ */
